<a href="https://colab.research.google.com/github/H2D2R2/Dz5-reworked-/blob/main/%D0%94%D0%BE%D0%BC%D0%B0%D1%88%D0%BD%D0%B5%D0%B5_%D0%B7%D0%B0%D0%B4%D0%B0%D0%BD%D0%B8%D0%B5_5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# (Исправленное) 

# Задание 1

Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

In [ ]:
M = 80
z = 1.96
sigma = 16                                                                      #исправил значение и наименование сигмы для наглядности 
n = 256

A0 = M - (z * sigma/np.sqrt(n))
A1 = M + (z * sigma/np.sqrt(n))                                                 #и добавил корень из объема выборки (раньше его нашел без формулы)

A0

78.04

In [ ]:
A1

81.96

Ответ: {78.04;81.96}





# Задание 2

В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95. 

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import warnings

a = [6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1]
a.sort()
a

[6.1, 6.1, 6.2, 6.3, 6.4, 6.7, 6.8, 6.9, 6.9, 7.5]

In [ ]:
sigma = np.std(a, ddof=1)
sigma

0.4508017549014448

In [ ]:
x = np.mean(a)
x

6.590000000000001

In [ ]:
t = 2.262
n = 10

In [ ]:
A0 = x - t*(sigma/np.sqrt(n))
A0

6.267538255912426

In [ ]:
A1 = x + t*(sigma/np.sqrt(n))
A1

6.912461744087575

Ответ: {6.26;6.91}

# Задание 3

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 кв.мм.

In [ ]:
mu0 = 17
mu = 17.5
D = 4
sigma = np.sqrt(D)                                                              #добавил для наглядности обозначения mu и mu0, сигму как корень из дисперсии 
n = 100

zr = (mu - mu0)/(sigma/np.sqrt(n))                                              #расчеты были неправильными, так как не было скобки в нижней части деления, исправил
zr                                                                               #добавил z расчетное 


2.5

In [ ]:

zt = 1.6                               #добавил z табличное  

zt < zr

True

Ответ: Теория верна на уровне значимости 5%

# Задание 4

Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

In [ ]:
x = [202, 203, 199, 197, 195, 201, 200, 204, 194, 190]                          #изменил обозначения и убрал ненужную сортировку массива
mu0 = 200
t = 3.25
n = 10

In [ ]:
sigma = np.std(x, ddof=1)
sigma

4.453463071962462

In [ ]:
mu = np.mean(x)
mu

198.5

In [ ]:
H1 = (mu - mu0)/(sigma/np.sqrt(n))                                              #поправил формулу
H1

-1.0651074037450896

In [ ]:
H0 = -3.25                                                                      #нашел табличное значение и сравнил с расчетным
H1 > H0

False

Ответ: Утверждение продавца верно на уровне значимости 1%